### ISSUE TITLE
Period (.) is not allowed / Source code from repo stopped matching the compiled exe packages?

### ISSUE THREAD

**Issue by mark99i** [usebruno/bruno#2081]
### I have checked the following:

- [X] I use the newest version of bruno.
- [X] I've searched existing issues and found nothing related to my issue.

### Describe the bug

I am using the golden version of bruno (bought it in pre-order) and have run into the problem that I cannot rename api methods to a name containing a dot. 
I also cannot create a new method containing the dot symbol. 
but at the same time, I can clone an existing method containing a dot symbol - there are no errors for this

I checked the main branch of the current repository and did not find any related commits, I also ran the main branch in my dev mode and also did not find this problem.
are you building exe and other packages not from this repository?

if this is the case, then there are doubts regarding the security of the code contained in the assembled packages. 
if this is related to paid subscription modules, then you can download them dynamically (from your server over https, for example)

### .bru file to reproduce the bug

none

### Screenshots/Live demo link

![image](https://github.com/usebruno/bruno/assets/46539611/258b06a7-3b83-4ded-af7c-f96edc42e1f0)
![image](https://github.com/usebruno/bruno/assets/46539611/d27bdfce-41b4-4d23-b078-13a8c7664ff0)

**Comment by gengurke** [comment]
I noticed the same/similar behavior.
I can create new Request with dots in the name but cannot rename them afterwards when they still contain a dot
**When _creating_ request:**
![image](https://github.com/user-attachments/assets/eb7d4f21-4ab7-4b08-9c10-32f256cb7e98)
This works fine


**When _editing_ request:**
![image](https://github.com/user-attachments/assets/799515f8-9429-41c3-9d07-4d7930d75a7c)
Cannot save even without change

**Comment by StephenBeirlaen** [comment]
Indeed, it appears that this condition is not in place in this repository.

This is what the code for the renaming step looks like in the debugger, when installed through snap:

```javascript
validationSchema: n6.object({
    name: n6.string()
        .matches(/^[^.]*$/, "Period (.) is not allowed") // <---- Added
        .min(1, "must be at least 1 character")
        .max(255, "must be 255 characters or less")
        .required("name is required"),
    filename: n6.string()
        .min(1, "must be at least 1 character")
        .max(255, "must be 255 characters or less")
        .required("name is required")
        .test("is-valid-name", function(t) {
            return !!ii(t) || this.createError({
                message: io(t)
            }
        )
    }).test("not-reserved", 'The file names "collection" and "folder" are reserved in bruno', t => !["collection", "folder"].includes(t))
}),
```

This does not match the code in https://github.com/usebruno/bruno/blob/526fcabffef8afe77274395a176385cf35f73819/packages/bruno-app/src/components/Sidebar/Collections/Collection/CollectionItem/RenameCollectionItem/index.js#L38-L52.

**Comment by Its-treason** [comment]
Yes, this will also be true for many other places in the source code. The released version of Bruno is proprietary, its source code is in another repo that is built upon the code from this repo and includes the Golden Edition features.

**Comment by mark99i** [comment]
@helloanoop @Pragadesh-45 @pooja-bruno 
Dear developers, can you explain why the dot is prohibited?
I'm using the Golden version.

**Comment by anusree-bruno** [comment]
@mark99i 
This is a regression we have a PR that fixes this, https://github.com/usebruno/bruno/pull/4373. This is currently under review.

**Comment by fractalf** [comment]
+1

There was an issue some time ago regarding the same thing, just `http://`  (urls).

Here's a tip: Allow the user to **name their requests** whatever. When you make the actual **filename** just escape all non-alphabetical characters with `-`. It really should be that easy

Bonus! With one line of code (or less) you can truncate multiple `-` to make it even nicer...

![Image](https://github.com/user-attachments/assets/a2f63540-c895-4832-ac34-db31e2d2b6d8)

**Comment by dverzolla** [comment]
Trying to create a request inside a collection, trying using "." and sometimes Bruno permits and other does not.

**Comment by solak02** [comment]
I’ve compared the .exe and .app from the 3.0.2 releases and built the app from tags/v3.0.2, and the version built from source doesn’t have the period lock. Tested on Windows and macOS. @anusree-bruno, is this intended, or is it a regression?

**Comment by anusree-bruno** [comment]
This seems to be fixed in the latest versions. Closing this as resolved

https://github.com/user-attachments/assets/2796eb5a-7279-4730-ae6b-02546e940153
